app/controllers/users_controller.rb

```diff
@@ -0,0 +1,17 @@
+class UsersController < ApplicationController
+  def index
+    fetch_current_user_details
+    render json: {
+      current_user: current_user,
+      current_user_roles: @current_user_roles,
+      users: User.all
+    }
+  end
+
+private
+
+  def fetch_current_user_details
+    user = current_user
```

Comment:
Updated.
I think that this approach is not the best one anyway because of the following:
1. I don't need all the users list right now (but definitely will a bit later).
2. I need only `current_user: current_user` and `current_user_roles: @current_user_roles` so far.
3. This solution propagates an issue during page refresh: the data are not parsed in time when a view needs the data.

I suggest to create dedicated method that will provide just exact data I need.